tools/libpico/CMakeLists.txt

```diff
@@ -1,216 +1,237 @@
+        pico_stdlib
+    )
+
+    target_link_libraries(ipv4-${cpu}
+        ${picow_link_libraries}
+    )
+
+    target_link_libraries(ipv4-big-${cpu}
+        ${picow_link_libraries}
+    )
+
+    add_library(ipv4-ipv6-${cpu} STATIC)
+    target_compile_definitions(ipv4-ipv6-${cpu} PUBLIC
+        LWIP_IPV6=1
+    )
+
+    add_library(ipv4-ipv6-big-${cpu} STATIC)
+    target_compile_definitions(ipv4-ipv6-big-${cpu} PUBLIC
+        __LWIP_MEMMULT=2
+        LWIP_IPV6=1
+    )
+
+    target_link_libraries(ipv4-ipv6-${cpu}
+        ${picow_link_libraries}
+    )
+        
+    target_link_libraries(ipv4-ipv6-big-${cpu}
+        ${picow_link_libraries}
+    )
+
+    if(${cpu} MATCHES "rp2040")
+        add_library(ipv4-bt-${cpu} STATIC)
+        target_compile_definitions(ipv4-bt-${cpu} PUBLIC
+            LWIP_IPV6=0
+        )
+
+        add_library(ipv4-bt-big-${cpu} STATIC)
+        target_compile_definitions(ipv4-bt-${cpu} PUBLIC
+            __LWIP_MEMMULT=2
+            LWIP_IPV6=0
+        )
+
+        set(picow_bt_link_libraries
+            pico_btstack_cyw43
+            pico_btstack_ble
+            pico_btstack_classic
+            pico_btstack_sbc_encoder
+            pico_btstack_sbc_decoder
+        )
+
+        target_link_libraries(ipv4-bt-${cpu}
+            ${picow_link_libraries}
+            ${picow_bt_link_libraries}
+        )
+
+        target_link_libraries(ipv4-bt-big-${cpu}
+            ${picow_link_libraries}
+            ${picow_bt_link_libraries}
+        )
+
+        add_library(ipv4-ipv6-bt-${cpu} STATIC)
+        target_compile_definitions(ipv4-ipv6-bt-${cpu} PUBLIC
+            LWIP_IPV6=1
+        )
+
+        add_library(ipv4-ipv6-bt-big-${cpu} STATIC)
+        target_compile_definitions(ipv4-ipv6-bt-big-${cpu} PUBLIC
+            __LWIP_MEMMULT=2
+            LWIP_IPV6=1
+        )
+
+        target_link_libraries(ipv4-ipv6-bt-${cpu}
+            ${picow_link_libraries}
+            ${picow_bt_link_libraries}
+        )
+
+        target_link_libraries(ipv4-ipv6-bt-big-${cpu}
+            ${picow_link_libraries}
+            ${picow_bt_link_libraries}
+        )
+    endif()
+
+    if(${cpu} MATCHES "rp2040")
+        set(tx ipv4-bt ipv4-ipv6-bt ipv4-bt-big ipv4-ipv6-bt-big)
+    elseif(${cpu} MATCHES "rp2350")
+        set(tx)
+    endif()
+    foreach(tgt pico ipv4 ipv4-ipv6 ipv4-big ipv4-ipv6-big ${tx})
+        add_custom_command(TARGET ${tgt}-${cpu} POST_BUILD
+            COMMAND ar d lib${tgt}-${cpu}.a stdio.c.obj stdio_uart.c.obj stdio_usb.c.obj stdio_usb_descriptors.c.obj pico_malloc.c.obj newlib_interface.c.obj
+            COMMAND ar d lib${tgt}-${cpu}.a btstack_flash_bank.c.obj  # Need to override with our own implementation
+            COMMAND cp lib${tgt}-${cpu}.a ../../../lib/${cpu}/lib${tgt}.a
+        )
+    endforeach()
+    add_custom_command(TARGET pico-${cpu} POST_BUILD
+            COMMAND cp ./generated/pico_base/pico/version.h ../../../include/${cpu}/pico_base/pico/.
```

Comment:
`config_autogen.h` is missing here. Also, I had to create the rp2350 directories by hand, maybe adding a `mkdir -p` here would solve that.

Reply:
Good catch!  Points out another issue.  I make the SDK build in unique subdirs, but the config_autogen.h is always put at {build root dir}/pico/... so it's not possilbe to build 2 chips in the same subdir.  I've modified my makefiles to do it sequentially in different dirs...